pkgs/development/interpreters/ruby/default.nix

```diff
@@ -17,7 +17,7 @@ let
   # Contains the ruby version heuristics
   rubyVersion = import ./ruby-version.nix { inherit lib; };
 
-  generic = { version, sha256 }: let
+  generic = { version, revision, sha256 }: let
```

Comment:
Could we fallback to the older behavior if `revision` is not set 🤔 ?

Reply:
I don't think that makes sense without more changes. The `generic` function is private to this file, so it will only be called from the section that defines `ruby_2_7`, `ruby_3_0`, and `ruby_3_1`. These callers of the `generic` function should always specify the `revision` of the tag they're fetching.

Ideally the `revision` would be attached to the `src` attribute, so when the `src` was overridden it would use the new value or fall back to `"HEAD"`, but I don't see an easy way to implement this.